### DISCUSSION TITLE
Receiving exception "The read operation timed out" using Python SDK

### DISCUSSION THREAD

**Discussion by hosseinmay** [deepgram/community#733] [General help] [answered]
### Which Deepgram product feature are you using?

Deepgram API - STT Pre-recorded

### Details

Hey Deepgram team, hope you're all doing well! 

I frequently (>50% of the time) receive a "The read operation timed out" error message when running a pre-recorded audio file through the Deepgram Python SDK. However, all these requests are shown as successful through the console, and I am also billed for their use. These audio files aren't that large (<11MB and <12 mins long), but they usually fail transcription and I have to retry sending them through, causing my bill to rack up. 

Here's a list of example request IDs:

- a8265e4e-fd36-4e52-9c19-1e53ccb76ed8
- 0efb8403-9c51-4d0e-b490-c0a0fe485284
- 791e691c-c937-4008-a11f-4ccbb2940f8a
- 931ff649-eb2b-44bd-990b-5a10fa4c3d6f
- 0fb083d3-8ca0-47df-93f2-39f63628f446
- ede46b2f-1d5d-436f-826d-63048fd7b157
- ddc0fa74-2359-40de-a44e-c23e04062a12
- 5c8eefeb-d2c5-4b49-a2e7-516f1920afb7

Thanks for your time,
Hossein

### If you are making a request to the Deepgram API, what is the full Deepgram URL you are making a request to?

/v1/listen?diarize=true&model=whisper-large&multichannel=true&paragraphs=true&smart_format=true

### If you are making a request to the Deepgram API and have a request ID, please paste it below:

a8265e4e-fd36-4e52-9c19-1e53ccb76ed8

### If possible, please attach your code or paste it into the text box.

``` python
def call_deepgram_api(audio_path):
    for i in range(0,5):
        logger.info(f"Try {i} at calling deepgram API")
        while True:
            try:
                config = DeepgramClientOptions(                    
                    options={"keepalive": "true"}  # have also tried without DeepgramClientOptions, doesn't seem to make a difference.
                )
                deepgram = DeepgramClient(DEEPGRAM_API_KEY, config)
        
                with open(audio_path, "rb") as file:
                    buffer_data = file.read()
        
                payload: FileSource = {
                    "buffer": buffer_data,
                }
        
                options = PrerecordedOptions(
                    diarize=True,
                    model="whisper-large", # TODO: test out different models, e.g. nova-2-meeting, whisper-large, nova-2, enhanced etc.
                    smart_format=True,
                    paragraphs=True,
                    multichannel=True                    
                )        
        
                response = deepgram.listen.prerecorded.v("1").transcribe_file(payload, options)                
                return response
        
            except Exception as e:
                logger.info(f"Exception: {e}")
            break
```

**Comment by team-deepgram** [comment]
Thanks for asking your question about Deepgram! If you didn't already include it in your post, please be sure to add as much detail as possible so we can assist you efficiently, such as:
- The `request_id` if you have a question about your requests or transcription responses.
- The features you used or the full `api.deepgram.com` URL you sent your request to, including parameters.
- Any code snippets you can share.

**Comment by davidvonthenen** [comment, marked as answer]
Hi @hosseinmay 

You just need to increase the timeout for larger files.
https://github.com/deepgram/deepgram-python-sdk/blob/main/examples/prerecorded/file/main.py#L51

The processing time also includes upload time. Depending on the options used, it make require more processing time.

**Reply by hosseinmay** [reply]
Perfect, thank you for your quick support!
